### PR TITLE
Explicitly close InputStream in RemoteFileFetcher

### LINKED ISSUE #313: Unable to delete lint file after reporting them back to Phabricator
I am running into an incredibly frustrating issue with Jenkins/Phabricator on Windows. After generating a `.phabricator-lint` file and reporting it back to Phabricator at the end of the build process, it becomes quite difficult to delete the file for the next run. At first, it was the `git reset` and `git clean` steps which were responsible for removing the lint file from the previous run. After git "deletes" the file, there is still a 0 B file left over that cannot be deleted without restarting the Jenkins service. If I manually delete the file from outside of Jenkins in between runs (before the SCM step tries to delete it), it will delete just fine.

My original workaround was to add the file to the `.gitignore` so that the `git clean` would not try to delete it, and the build process would just overwrite it. This worked well until someone submitted a diff that branched from before the change to `.gitignore`, triggering an attempt to delete it again from `git clean`. If I disable the step where it reports the lint issues back to phabricator, `git clean` can remove the file without issues.

I have also tried deleting the file before the SCM step with the Pre SCM Buildstep plugin, but that, too, fails to delete the file properly, leaving it in the same broken state.

Using Process Explorer, I have found that the `java.exe` under Jenkins has a file handle open to the `.phabricator-lint` file after the job finishes. If I force close the handle in between jobs, but leave the old lint file there, the next time the job is run, it can delete the file as it should. So it appears that there is a file handle that remains open after reporting the lint issues to Phabricator, which is somehow preventing the file from being removed _only from within Jenkins and its child processes_, but not from outside of Jenkins; as I mentioned above, I can delete the file with Explorer between jobs, even it the file handle is open.

Does anyone have any clues as to what is going on? Anything that can be done to close that file handle?

### PR DESCRIPTION
**Pull request by youngmit**
This is an attempt to fix #313.

### CODE REVIEW

**Comment by youngmit** [comment]
Tested this against the issues I was seeing in #313, and it appears to have done the trick. Can repeatedly run the job without error.

File: src/main/java/com/uber/jenkins/phabricator/RemoteFileFetcher.java [hunk 1]
```diff
@@ -80,7 +81,9 @@ public String getRemoteFile() throws InterruptedException, IOException {
             maxLength = (int) source.length();
         }
         byte[] buffer = new byte[maxLength];
-        IOUtils.read(source.read(), buffer);
+        InputStream stream = source.read();
+        IOUtils.read(stream, buffer);
+        stream.close();
```
Comment (prax-uber) [inline src/main/java/com/uber/jenkins/phabricator/RemoteFileFetcher.java]: Just a thought. You may want to close the stream in `finally` block and perform a basic check. Something like - 
```
try {
    IOUtils.read(stream, buffer);
} finally {
    try { if (stream != null) stream.close(); } catch(IOException e) { /* ignore */ }      
}
```

Comment (youngmit) [reply src/main/java/com/uber/jenkins/phabricator/RemoteFileFetcher.java]: Thanks for the recommendation. New to Java...

**Comment by youngmit** [comment]
@prax-uber could I get your eyes on this one again? No rush, just want to make sure it doesn't slip through the cracks... Thanks!

**Comment by prax-uber** [comment]
hey @youngmit. Your changes looks good to me but I am not a committer/merger on this repo.
I think one of the existing mergers/committers probably should look into this to move the needle. Thx!

**Comment by youngmit** [comment]
@prax-uber Thanks for the update. Do you know who does have permissions to merge this? I don't even know where to look.

**Pull request merged**